### DISCUSSION TITLE
Tree: selection mode support

### DISCUSSION THREAD

**Discussion by jasdom** [iTwin/iTwinUI#2027] [Ideas]
Hi, we are replacing the old tree component with the one provided by `@itwin/itwinui-react`, and the previous implementation supported multiple selection modes that allowed the use of ctrl, shift and drag to select/deselect multiple nodes. Are there any plans to add support for a similar feature?

If support for such modes is not planned, we would at least be interested in shift+clicking nodes to select all nodes between the previous selection and the node that was shift+clicked.

**Comment by mayank99** [comment]
There are no immediate plans, but I think in the future we could look into supporting at least a subset of what you requested.

Recently we got another question about supporting similar functionality in `TransferList`: https://github.com/iTwin/iTwinUI/discussions/2018

It would be nice to centralize this kind of logic to be used in `Table`, `TransferList` and `Tree`.
